PWGMM/Mult/Tasks/dndeta-hi.cxx

```diff
@@ -318,551 +332,567 @@ struct MultiplicityCounter {
-            }
-            registry.fill(HIST("Tracks/ProcessMCV0Counting/hV0Mass"), Double_t(kINEL), Double_t(kAntilambda), v0.mAntiLambda());
-            if (1.11 < v0.mAntiLambda() && v0.mAntiLambda() < 1.12 && abs(v0.yLambda()) < rapidity) {
-              registry.fill(HIST("Tracks/ProcessMCV0Counting/hV0Count"), Double_t(kINEL), Double_t(kAntilambda), Double_t(kMasscut));
-              registry.fill(HIST("Tracks/ProcessMCV0Counting/hV0DauEta"), Double_t(kINEL), Double_t(kPositive), Double_t(kAntilambda), pTrack.eta());
-              registry.fill(HIST("Tracks/ProcessMCV0Counting/hV0DauEta"), Double_t(kINEL), Double_t(kNegative), Double_t(kAntilambda), nTrack.eta());
-            }
-          }
+    auto genz = mcCollision.posZ();
+    Bool_1d bevtc(kECend, false);
+    bevtc[kINEL] = true;
+    for (auto& particle : mcParticles) {
+      if (!particle.isPhysicalPrimary())
+        continue;
+      auto p = pdg->GetParticle(particle.pdgCode());
+      if (p != nullptr) {
+        if (std::abs(p->Charge()) >= 3) {
+          if (std::abs(particle.eta()) < 1)
+            bevtc[kINELg0] = true;
         }
       }
-
-      auto particles = mcSample->sliceByCached(aod::mcparticle::mcCollisionId, mcCollision.globalIndex(), cache);
-      auto tracks = lsample->sliceByCached(aod::track::collisionId, collision.globalIndex(), cache);
-      tracks.bindExternalIndices(&mcParticles);
-
-      usedTracksIds.clear();
-      for (auto& track : atracks) {
-        auto ttrack = track.track_as<soa::Filtered<LabeledTracksEx>>();
-        usedTracksIds.emplace_back(ttrack.globalIndex());
-        if (ttrack.has_mcParticle()) {
-          for (auto MotherIDs = ttrack.mcParticle_as<Particles>().mothersIds().front(); MotherIDs <= ttrack.mcParticle_as<Particles>().mothersIds().back(); MotherIDs++) {
-            auto mother = mcParticles.rawIteratorAt(MotherIDs);
-            auto pdg_mother = mother.pdgCode();
-            if (pdg_mother == 310) {
-              registry.fill(HIST("Tracks/ProcessMCV0Counting/hMotherV0Count"), Double_t(kINEL), Double_t(kK0short));
-            }
-            if (pdg_mother == 3122) {
-              registry.fill(HIST("Tracks/ProcessMCV0Counting/hMotherV0Count"), Double_t(kINEL), Double_t(kLambda));
-            }
-            if (pdg_mother == -3122) {
-              registry.fill(HIST("Tracks/ProcessMCV0Counting/hMotherV0Count"), Double_t(kINEL), Double_t(kAntilambda));
-            }
-          }
-        } else {
-          // when secondary
-        }
+    }
+    // cout<<"INEL? "<<bevtc[kINEL]<<"  INEL>0? "<<bevtc[kINELg0]<<endl;
+    for (auto ievtc = 1u; ievtc < kECend; ievtc++) {
+      if (bevtc[ievtc])
+        registry.fill(HIST("hgenzvtx"), Double_t(ievtc), genz, -1.0);
+    }
+    Int_t pid = 0;
+    for (auto& particle : mcParticles) {
+      auto p = pdg->GetParticle(particle.pdgCode());
+      if (std::abs(particle.pdgCode() == 310 && fabs(particle.eta()) < 0.5) && fabs(genz) < 10)
```

Comment:
The parentheses of `std::abs` are closed incorrectly, it should be `std::abs(particle.pdgCode()) ==`.

Reply:
Yes, thanks for the exact advise